src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,260 @@
+package ui.components.pickers;
+
+import backend.resource.TurboLabel;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to represent the state of the label picker. In addition to representing a state,
+ * it contains the logic that handles every state transition
+ */
+public class LabelPickerState {
+
+    private Set<String> initialLabels;
+    private List<String> addedLabels;
+    private List<String> removedLabels;
+    private List<String> matchedLabels;
```

Comment:
Shouldn't we use `List<TurboLabel>` instead of `List<String>`? Wouldn't that avoid lot of back and forth conversions? It's more OOP to keep things as objects and convert to primitive forms only when necessary.